### DISCUSSION TITLE
how to use DNS module

### DISCUSSION THREAD

**Discussion by lcinrain** [zmap/zmap#844] [Q&A] [answered]
I want to scan a set of candidate IPv4 DNS server with their IP addresses in a file. I want to find which addresses can respond to the DNS query. What's the command?   I use <sudo zmap --probe-module=dns --output-module=json -p 53 -i eth0 <resolvers.txt -o a.json>. Is that right? The candidate DNS server IP addresses are in the file resolvers.txt.

**Comment by droe** [comment, marked as answer]
You're missing `--probe-args` specifying which rrtype and question to use in the DNS query.  See the source code for details:  https://github.com/zmap/zmap/blob/main/src/probe_modules/module_dns.c#L11-L37
